services/migrations/gitea_uploader.go

```diff
@@ -217,11 +220,17 @@ func (g *GiteaLocalUploader) CreateMilestones(milestones ...*base.Milestone) err
 func (g *GiteaLocalUploader) CreateLabels(labels ...*base.Label) error {
 	lbs := make([]*issues_model.Label, 0, len(labels))
 	for _, label := range labels {
+		// We must validate color here:
+		if !issues_model.LabelColorPattern.MatchString("#" + label.Color) {
+			log.Warn("Invalid label color: #%s for label: %s in migration to %s/%s", label.Color, label.Name, g.repoOwner, g.repoName)
```

Comment:
TBH all of the logs need to have the context in them - which is simply `g` either logged as `%s` or `%-v`

However, this is less of an issue because all of our logs log the [PID] so you can figure of what the context is from that.

Reply:
Actually, we don't have the baseURL here.

This is in the Uploader which does not have access to the Downloader - this is a design issue which will need to be fixed.